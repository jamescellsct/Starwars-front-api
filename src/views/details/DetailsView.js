import React, { Component } from "react";
import { Link, withRouter } from "react-router-dom";
import { connect } from "react-redux";
import { Paper, Button } from "@material-ui/core";
import LoadingComponent from "../../components/details/LoadingComponent";
import DetailsComponent from "../../components/details/detailsComponent";

class DetailsView extends Component {
  render() {
    let data = this.props.location.state.find(
      item => item.idItem === +this.props.match.params.id
    );
    const resource = this.props.match.params.object;

    return data ? (
      <div
        style={{
          position: "fixed",
          width: "100%",
          height: "100%",
          zIndex: "-10",
          background: "#343a40"
        }}
      >
        <div className="d-flexflex-column">
          <div className="col-12 bg-dark">
            <label className="lead" style={{ color: "white" }}>
              <strong>Details Page: {resource} </strong>
            </label>
          </div>
          <div
            className="d-flex justify-content-center align-items-stretch"
            style={{ height: 600 }}
          >
            <Paper
              elevation={1}
              className="w-50 h-100 d-flex flex-column align-items-center"
              style={{ margin: 20 }}
            >
              <div
                className="d-flex h-50 flex-column align-items-center justify-content-start"
                style={{ marginTop: 10 }}
              >
                <div>
                  <img
                    alt=""
                    src={data.avatar.replace("/40?", "/160?")}
                    className="img-fluid border border-ligth rounded-circle"
                  />
                </div>
                <div>
                  <label className="display-4">{data.name}</label>
                </div>
              </div>
              <DetailsComponent data={data} resource={resource} />
              <div style={{ marginBottom: 20 }}>
                <Link
                  to={{
                    pathname: `/${this.props.match.params.object}`,
                    state: this.props.location.state
                  }}
                  style={{ textDecoration: "none" }}
                >
                  <Button variant="contained">To Back</Button>
                </Link>
              </div>
            </Paper>
          </div>
        </div>
      </div>
    ) : (
      <div
        style={{
          position: "fixed",
          width: "100%",
          height: "100%",
          zIndex: "-10",
          background: "#343a40"
        }}
      >
        <LoadingComponent />;
      </div>
    );
  }
}

const mapStateToProps = state => ({
  state
});

export default withRouter(connect(mapStateToProps)(DetailsView));
