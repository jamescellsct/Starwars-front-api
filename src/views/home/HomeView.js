import React, { Component } from "react";

import swlogo from "../../images/swlogo.png";
import planet from "../../images/_planet.png";
import starship from "../../images/_starship.png";
import vehicle from "../../images/_vehicle.png";
import people from "../../images/_people.png";

import MenuComponent from "../../components/home/MenuComponent";
import "./HomeView.css";

class HomeView extends Component {
  render() {
    return (
      <div
        id="container"
        style={{
          position: "fixed",
          width: "100%",
          height: "100%",
          zIndex: "-10",
          backgroundColor: "#343a40"
        }}
      >
        <div id="subContainer" className="d-block bg-light" />
        <div
          id="head"
          className="d-flex h-50 justify-content-center align-items-center"
          style={{ backgroundColor: "#2a2f35" }}
        >
          <img src={swlogo} alt="" style={{ height: "auto", width: 300 }} />
        </div>
        <div id="body" className="d-flex flex-row justify-content-center h-50">
          <div
            id="menu"
            className="d-flex bg-dark w-75 h-100 flex-row align-items-center"
          >
            <MenuComponent itemName="Peoples" itemImage={people} />
            <MenuComponent itemName="Planets" itemImage={planet} />
            <MenuComponent itemName="Starships" itemImage={starship} />
            <MenuComponent itemName="Vehicles" itemImage={vehicle} />
          </div>
        </div>
      </div>
    );
  }
}

export default HomeView;
