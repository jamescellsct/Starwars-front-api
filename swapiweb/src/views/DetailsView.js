import React, { Component } from 'react';
import { Link } from 'react-router-dom';

class DetailsView extends Component {

    render() {
        return(
            <Link to='/'>
                <h4>Details Page</h4>
            </Link>
        );
    }
}

export default DetailsView;