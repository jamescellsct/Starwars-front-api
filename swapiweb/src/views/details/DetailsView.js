import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { detailsRequest } from '../../services/swapi_connect';
import {
    Paper,
    Button,
    Divider
} from '@material-ui/core';
import LoadingComponent from '../../components/LoadingComponent';

class DetailsView extends Component {
    constructor() {
        super();

        this.state = {
            data: null,
        }
    }

    handleState() {
        const {itemName} = this.props.match.params;
        const {state} = this.props.location;

        return itemName === 'people' ? state.peoples :
            itemName === 'planets' ? state.planets :
            itemName === 'starships' ? state.starships : state.vehicles;
    }

    componentDidMount() {
        const { itemName, id } = this.props.match.params;
        const {object} = this.handleState();

        let item = itemName === 'people' ? 'peoples' : itemName;

        this.props.location.state && item.toUpperCase() === object.toUpperCase() ? (() => {

            let { results } = this.handleState();

            console.log(results);
            

            let res = results.filter((item) => {
                return item.idItem === id;
            });

            console.log(res);
            

            this.setState({ data: res[0] });
        })() : detailsRequest(itemName, id).then((value) => {
            const { data } = value;
            this.setState({ data });
        });

    }

    render() {

        const { data } = this.state;
        const { itemName } = this.props.match.params;
        const { state } = this.props.location;

        return data ? (
            <div style={{
                position: "fixed",
                width: "100%",
                height: "100%",
                zIndex: "-10",
                background: "#343a40"
            }}>
                <div className="d-flexflex-column">
                    <div className="col-12 bg-dark">
                        <label className="lead" style={{ color: 'white' }}>
                            <strong>Details Page: {itemName} </strong>
                        </label>
                    </div>
                    <div className="d-flex justify-content-center align-items-stretch" style={{ height: 600 }}>
                        <Paper elevation={1} className="w-50 h-100 d-flex flex-column align-items-center" style={{ margin: 20 }}>
                            <div className="d-flex h-50 flex-column align-items-center justify-content-start" style={{ marginTop: 10 }}>
                                <div>
                                    <img alt="" src={data.avatar.replace('40?', '120?')} className="img-fluid border border-ligth rounded-circle" />
                                </div>
                                <div>
                                    <label className="display-4">
                                        {data.name}
                                    </label>
                                </div>
                            </div>
                            <div className="d-flex w-100">
                                <Divider variant="inset" />
                            </div>
                            {
                                itemName === 'planets' ?
                                    <div className="d-flex flex-row h-75 w-75 align-items-start justify-content-between">
                                        <div className="d-flex flex-column">
                                            <label className="lead">
                                                <strong>Terrain:</strong> {data.terrain}
                                            </label>
                                            <label className="lead">
                                                <strong>Gravity:</strong> {data.gravity}
                                            </label>
                                            <label className="lead">
                                                <strong>Diameter:</strong> {data.diameter}
                                            </label>
                                            <label className="lead">
                                                <strong>Climate:</strong> {data.climate}
                                            </label>
                                            <label className="lead">
                                                <strong>Popularion:</strong> {data.population}
                                            </label>
                                            <label className="lead">
                                                <strong>Orbital Period:</strong> {`${data.orbital_period} Days`}
                                            </label>
                                            <label className="lead">
                                                <strong>Rotation Period:</strong> {`${data.rotation_period} Days`}
                                            </label>
                                            <label className="lead">
                                                <strong>Surface Water:</strong> {data.surface_water}
                                            </label>
                                        </div>
                                    </div> :

                                    itemName === 'vehicles' ?
                                        <div className="d-flex flex-row h-75 w-75 align-items-start justify-content-between">
                                            <div className="d-flex flex-column">
                                                <label className="lead">
                                                    <strong>Vehicle Class:</strong> {data.vehicle_class}
                                                </label>
                                                <label className="lead">
                                                    <strong>Model:</strong> {data.model}
                                                </label>
                                                <label className="lead">
                                                    <strong>Manufacturer:</strong> {data.manufacturer}
                                                </label>
                                                <label className="lead">
                                                    <strong>Cargo Capacity:</strong> {data.cargo_capacity} Pounds
                                        </label>
                                                <label className="lead">
                                                    <strong>Passengers:</strong> {data.passengers} Persons
                                        </label>
                                                <label className="lead">
                                                    <strong>Length:</strong> {data.length} Meters
                                        </label>
                                                <label className="lead">
                                                    <strong>Consumables:</strong> {data.consumables}
                                                </label>
                                                <label className="lead">
                                                    <strong>Max Atmosphering Speed:</strong> {data.max_atmosphering_speed}
                                                </label>
                                            </div>
                                        </div> :

                                        itemName === 'starships' ?
                                            <div className="d-flex flex-row h-75 w-75 align-items-start justify-content-between">
                                                <div className="d-flex flex-column">
                                                    <label className="lead">
                                                        <strong>Starship Class:</strong> {data.starship_class}
                                                    </label>
                                                    <label className="lead">
                                                        <strong>Model:</strong> {data.model}
                                                    </label>
                                                    <label className="lead">
                                                        <strong>Manufacturer:</strong> {data.manufacturer}
                                                    </label>
                                                    <label className="lead">
                                                        <strong>Cargo Capacity:</strong> {data.cargo_capacity} Pounds
                                        </label>
                                                    <label className="lead">
                                                        <strong>Passengers:</strong> {data.passengers} Persons
                                        </label>
                                                    <label className="lead">
                                                        <strong>Length:</strong> {data.length} Meters
                                        </label>
                                                    <label className="lead">
                                                        <strong>Consumables:</strong> {data.consumables}
                                                    </label>
                                                    <label className="lead">
                                                        <strong>Max Atmosphering Speed:</strong> {data.max_atmosphering_speed}
                                                    </label>
                                                </div>
                                            </div> :

                                            itemName === 'people' ?
                                                <div className="d-flex flex-row h-75 w-75 align-items-start justify-content-between">
                                                    <div className="d-flex flex-column">
                                                        <label className="lead">
                                                            <strong>Gender:</strong> {data.gender}
                                                        </label>
                                                        <label className="lead">
                                                            <strong>Eye Color:</strong> {data.eye_color}
                                                        </label>
                                                        <label className="lead">
                                                            <strong>Hair Color:</strong> {data.hair_color}
                                                        </label>
                                                        <label className="lead">
                                                            <strong>Height:</strong> {data.height} Centimeters
                                        </label>
                                                        <label className="lead">
                                                            <strong>Skin Color:</strong> {data.skin_color}
                                                        </label>
                                                        <label className="lead">
                                                            <strong>Mass:</strong> {data.mass} pounds
                                        </label>
                                                        <label className="lead">
                                                            <strong>Birth Year:</strong> {data.birth_year}
                                                        </label>
                                                        <label className="lead">
                                                            <strong>Vehicles:</strong> n/a
                                        </label>
                                                    </div>
                                                </div> : null
                            }
                            <div style={{ marginBottom: 20 }}>
                                <Link to={{
                                    pathname: `/${this.props.match.params.itemName}/table`,
                                    state
                                }} style={{ textDecoration: 'none' }}>
                                    <Button variant="contained">
                                        To Back
                                </Button>
                                </Link>
                            </div>
                        </Paper>
                    </div>
                </div>
            </div>
        ) :
            <div style={{
                position: "fixed",
                width: "100%",
                height: "100%",
                zIndex: "-10",
                background: "#343a40"
            }}>
                <LoadingComponent />;
        </div>
    }
}

export default DetailsView;