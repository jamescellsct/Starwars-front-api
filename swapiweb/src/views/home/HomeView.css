.paper {
  margin-top: 0px !important;
  transition-property: all !important;
  transition: 500ms !important;
}

.paper:hover {
  margin-top: -100px !important;
  transition: 200ms !important;
  cursor: pointer !important;
}
