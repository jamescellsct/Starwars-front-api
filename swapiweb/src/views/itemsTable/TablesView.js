import React, { Component } from 'react';
import planet from '../../images/_planet.png';
import starship from '../../images/_starship.png';
import vehicle from '../../images/_vehicle.png';
import people from '../../images/_people.png';
import swlogo from '../../images/swlogo.png';
import { Paper, Button, CircularProgress } from '@material-ui/core';
import { Link } from 'react-router-dom';
import { swapiRequest } from '../../services/swapi_connect';
import PaginationComponent from '../../components/itemsTable/PaginationComponent';
import TableComponent from '../../components/itemsTable/TableComponent';

class TablesView extends Component {
    constructor() {
        super();

        this.state = {
            stateNavigation: null,
            statePagination: null,
            loading: false
        }
    }

    getImage() {
        return this.props.match.params.object === 'planets' ? planet :
            this.props.match.params.object === 'starships' ? starship :
                this.props.match.params.object === 'vehicles' ? vehicle : people;
    }

    handleState(itemName) {
        const { state } = this.props.location;

        return itemName === 'people' ? state.peoples :
            itemName === 'planets' ? state.planets :
                itemName === 'starships' ? state.starships : state.vehicles;
    }

    getStateNavigation() {
        return this.state.stateNavigation || this.props.location.state;
    }

    goToPage(url = null, page = null) {
        let object = this.props.match.params.object;
        object = object === 'peoples' ? 'people' : object;

        const stateNavigation = this.getStateNavigation();

        this.setState({ loading: true, stateNavigation });

        swapiRequest(url, page, object).then((value) => {
            value.titlesTable = value.object === 'Peoples' ? ['Name', 'Gender', 'Eyes Color'] :
                                value.object === 'Planets' ? ['Name', 'Climate', 'Orbital Peroid'] :
                                value.object === 'Starships' ? ['Name', 'Model', 'Starship Class'] : ['Name', 'Model', 'Passengers'];
            let state = this.state.stateNavigation;

            value.object === 'Peoples' ? state.peoples = value :
                value.object === 'Planets' ? state.planets = value :
                value.object === 'Starships' ? state.starships = value: state.vehicles = value;
            
            this.setState({statePagination: value, loading: false, stateNavigation: state});
        })
    }

    render() {

        return (
            <div style={{
                position: "fixed",
                width: "100%",
                height: "100%",
                zIndex: "-10",
                backgroundColor: "#343a40"
            }}  >
                <div className="d-flex p-1 m-1 justify-content-between align-items-center">
                    <Paper elevation={5} className="d-flex justify-content-center align-items-center p-2">
                        <img src={this.getImage()} alt="" style={{ height: 90, width: "auto" }} />
                    </Paper>

                    <img src={swlogo} alt="" style={{ height: 100, width: "auto" }} />
                </div>
                <div className="d-block bg-light h-75 align-items-center justify-content-center">
                    <div className="d-flex bg-light h-100" style={{ borderRadius: 8 }}>
                        {
                            this.state.loading ?
                                <div className="d-flex flex-grow-1 justify-content-center align-content-center" >
                                    <div className="d-flex align-self-center flex-column align-items-center justify-content-center">
                                        <label className="p-4 lead">
                                            Data table in progress ...
                                         </label>
                                        <CircularProgress color="inherit" />
                                    </div>
                                </div>
                                : <TableComponent statePagination={this.state.statePagination} object={this.getStateNavigation()} itemName={this.props.match.params.object} />
                        }
                    </div>
                    <div className="h-25 w-50 pt-3 pl-3">
                        <PaginationComponent data={this.state.statePagination || this.handleState()} onSelected={(url, page) => this.goToPage(url, page)} />
                    </div>
                </div>
                <div className="d-block">
                    <div className="d-flex justify-content-center pt-3">
                        <Link to={{
                            pathname: '/',
                            state: this.getStateNavigation()
                        }} style={{ textDecoration: 'none', textDecorationStyle: 'none', color: '#777' }}>
                            <Button variant="contained" color="inherit">
                                <div className="lead">
                                    Voltar
                            </div>
                            </Button>
                        </Link>
                    </div>
                </div>
            </div>
        );
    }
}

export default TablesView;