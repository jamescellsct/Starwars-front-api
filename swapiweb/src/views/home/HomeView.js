import React, { Component } from 'react';
import swlogo from '../../images/swlogo.png';
import planet from '../../images/_planet.png';
import starship from '../../images/_starship.png';
import vehicle from '../../images/_vehicle.png';
import people from '../../images/_people.png';
import { swapiRequest } from '../../services/swapi_connect';
import MenuComponent from '../../components/home/MenuComponent';
import './HomeView.css';

class HomeView extends Component {
    constructor() {
        super();

        this.state = {
            peoples: null,
            planets: null,
            starships: null,
            vehicles: null,
        }
    }

    componentDidMount() {

        const { state } = this.props.location;

        state ? this.setState(state) :

            swapiRequest().then((peoples) => {
                peoples.titlesTable = ['Name', 'Gender', 'Eyes Color'];
                swapiRequest(null, null, 'planets').then((planets) => {
                    planets.titlesTable = ['Name', 'Climate', 'Orbital Peroid'];
                    swapiRequest(null, null, 'starships').then((starships) => {
                        starships.titlesTable = ['Name', 'Model', 'Starship Class'];
                        swapiRequest(null, null, 'vehicles').then((vehicles) => {
                            vehicles.titlesTable = ['Name', 'Model', 'Passengers'];
                            this.setState({ peoples, planets, starships, vehicles, isDone: true });;
                        });
                    });
                });
            });
    }

    render() {
        return (
            <div style={{
                position: "fixed",
                width: "100%",
                height: "100%",
                zIndex: "-10",
                backgroundColor: "#343a40"
            }}>
                <div className="d-block bg-light">

                </div>
                <div className="d-flex h-50 justify-content-center align-items-center" style={{ backgroundColor: '#2a2f35' }}>
                    <img src={swlogo} alt="" style={{ height: 'auto', width: 300 }} />
                </div>
                <div className="d-flex flex-row justify-content-center h-50">
                    <div className="d-flex bg-dark w-75 h-100 flex-row align-items-center">
                        <MenuComponent itemName='Peoples' itemImage={people} object={this.state} />
                        <MenuComponent itemName='Planets' itemImage={planet} object={this.state} />
                        <MenuComponent itemName='Starships' itemImage={starship} object={this.state} />
                        <MenuComponent itemName='Vehicles' itemImage={vehicle} object={this.state} />
                    </div>
                </div>
            </div>
        );
    }
}

export default HomeView;