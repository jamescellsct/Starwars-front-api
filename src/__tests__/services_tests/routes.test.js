import React from "react";
import { shallow, mount } from "enzyme";
import { Route, MemoryRouter } from "react-router-dom";

import Routes from "../../services/routes";

import DetailsView from "../../views/details/DetailsView";
import HomeView from "../../views/home/HomeView";
import TablesView from "../../views/itemsTable/TablesView";

let pathMap = {};
describe("Testing Routes Service", () => {
  beforeAll(() => {
    const component = shallow(<Routes />);
    pathMap = component.find(Route).reduce((pathMap, route) => {
      const routeProps = route.props();
      pathMap[routeProps.path] = routeProps.component;
      return pathMap;
    }, {});
  });

  it("Should render correctly HomeView component to route '/'", () => {
    const component = mount(
      <MemoryRouter initialEntries={["/"]}>
        <Routes />
      </MemoryRouter>
    );

    expect(component.find(HomeView)).toHaveLength(1);
    expect(pathMap["/"]).toBe(HomeView);
    expect(pathMap["/"]).toMatchSnapshot();
  });

  it("Should render correctly TablesView component to route '/:object'", () => {
    expect(pathMap["/:object"]).toBe(TablesView);
    expect(pathMap["/:object"]).toMatchSnapshot();
  });

  it("Should render correctly DetailsView component to route '/:object/:id'", () => {
    expect(pathMap["/:object/:id"]).toBe(DetailsView);
    expect(pathMap["/:object/:id"]).toMatchSnapshot();
  });
});
