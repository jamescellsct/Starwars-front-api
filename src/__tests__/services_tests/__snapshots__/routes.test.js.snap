// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Routes Service Should render correctly DetailsView component to route '/:object/:id' 1`] = `[Function]`;

exports[`Testing Routes Service Should render correctly HomeView component to route '/' 1`] = `[Function]`;

exports[`Testing Routes Service Should render correctly TablesView component to route '/:object' 1`] = `[Function]`;
